components/nearby-diagram.js

```diff
@@ -382,8 +416,11 @@ export default function NearbyDiagram({
                       y={scaleForGenePositionY - tickHeight - 5}
                       textAnchor="middle"
                     >
-                      {Math.floor(
-                        (index * tickWidth) / scaleForGene + geneScaleStartTick
+                      {getGeneScaleLabel(
+                        index,
+                        tickWidth,
+                        scaleForGene,
+                        geneScaleStartTick
```

Comment:
To continue from the previous comment, an advantage to assigning those functions to a variable and conditionally rendering based on the special invalid value, is that you can also make the `<text>` element render only when needed.